.github/workflows/ci.yml

```
@@ -174,4 +174,68 @@ jobs:
           file: ./coverage.xml
           flags: unittests
           name: codecov-umbrella
-          fail_ci_if_error: true
\ No newline at end of file
+          fail_ci_if_error: true
+
+  CLI-Compatibility:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        typer-version: ["0.16.0", "0.17.0", "0.18.0", "0.19.2"]
+        click-version: ["8.1.0", "8.2.0", "8.3.0"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Install Dependencies with specific typer and click versions
+        run: |
+          # Setup Virtual Environment
+          python3 -m venv ./.venv
+          source .venv/bin/activate
+          
+          # Install package with specific typer and click versions
+          pip install typer==${{ matrix.typer-version }} click==${{ matrix.click-version }}
+          pip install .[dev]
```

```suggestion
          if [ "${{ matrix.python-version }}" == "3.9" ]; 
            then
              pip install .[dev]
            else
              poetry install --extras dev
          fi
```
For python>=3.9, we should use poetry to install so that we get the versions of our other dependencies from the poetry.lock file.